Essence of Calculus Chapter 3 (#134)
3b1b/3Blue1Brown.com: public/content/lessons/2017/derivative-formulas-geometrically/index.mdx

$$
(x + dx)^3 = x^3 + \color{#fc6255} 3x^2 \color{black} dx \color{#AAAAAA} + 3xdx^2 + dx^3
$$

So the derivative of $x^3$, the rate at which $x^3$ changes per unit of change in $x$, is $3x^2$. Looking at the graph, this means the slope of the graph of $x^3$ at each point $x$ is exactly $3x^2$.

<Figure
  image="figures/6.46.svg"
  video="figures/6.22-6.46.mp4"
  show="video" 
/>

Graphical intuition with slope can tell us why this derivative is high on the left, $0$ at the origin, and high on the right, but just thinking in terms of graphs would not land us on the precise quantity $3x^2$. For that, we had to take a much more direct look at the actual meaning of the function.

### Derivative of $f(x) = x^n$

<!-- TODO: incorporate text into expressions-->

> In practice, you wouldn’t necessarily think of the square every time you’re taking a derivative of $x^2$, nor would you necessarily think of the cube when taking a derivative of $x^3$. Instead, thinking like a mathematician, can we generalize this approach to see if a pattern emerges? And maybe, if we are lucky, we invent a tool that we can use to find the derivative of any old polynomial.

Reviewer: ```suggestion
In practice, you wouldn’t necessarily think of the square every time you’re taking a derivative of $x^2$, nor would you necessarily think of the cube when taking a derivative of $x^3$. Instead, thinking like a mathematician, can you generalize this approach to see if a pattern emerges? Can you invent a tool to find the derivative of any polynomial?
```

<!-- TODO: link to binomial expansion -->

Focusing on the rate in change introduced by the very small nudge, highlighted in red, and ignoring expressions containing $dx^2$, the pattern that emerges is what's known in the business as the "power rule". Given a monomial function raised to some power, $n$, applying the rule gives us the derivative of the function.

<Figure
  image="figures/7.32.svg" 
  caption="Power Rule Definition"
/>

Even though in practice you will find yourself performing this derivative quickly and symbolically, imagining that exponent hopping down to the front, every now and then it’s nice to step back and remember why this rule works. Not just because it’s pretty, and not just because it helps to remind us that math actually makes sense and isn’t just a pile of formulas to memorize, but because it flexes that very important muscle of thinking about derivatives in terms of tiny nudges.

<Question
  question="What is the derivative of the function $f(x)= 3x^2$?"
  choice1="$2x$"
  choice2="$5x$"
  choice3="$3x$"
  choice4="$6x$"
  answer={4}
>

> This question can be modeled and solved as a question of change in area. For example, nudging the input of the function by $dx$ gives the expression $3(x+d x)^{2}$ which expands to the expression $3(x + dx)( x + dx)$. I've drawn the area below where the $3$ scales one of the sides.

Reviewer: ```suggestion
You certainly don't need to visualize this to answer the question. In fact, in practice, it would be absurd to do so. Nevertheless, continuing with the playful and visual spirit of this lesson, it's fun to think of a few ways to do just that.

This question can be modeled and solved as a question of change in area. For example, nudging the input of the function by $dx$ gives the expression $3(x+d x)^{2}$ which expands to the expression $3(x + dx)( x + dx)$. I've drawn the area below where the $3$ scales one of the sides.
```

<Figure

>   image="figures/power-rule-example.svg" 

Reviewer: I might opt for the 3 copies of the (square + nudged area) setup, just because it better emphasizes why multiplying the function by 3 just multiplied the derivative by 3.

Author: I agree

  caption="Derivative of $f(x) = 3x^{2}$ modeled as area"
/>

Looking at the figure we can see the rate of the change of the function represented by the new area of $6$ thin rectangles with area $x \cdot dx$. Alternatively, the question can be modeled as volume, where instead of scaling one of the sides, the $3$ extends the geometry into another dimension.

<Figure
  image="figures/3x^2-volume-model.svg" 
  caption="Derivative of $f(x) = 3x^{2}$ modeled as volume"
/>

This gives us the equivalent derivative, as the new volume of the two rectangular prisms is equal to $3 \cdot x \cdot dx + 3 \cdot x \cdot dx = 6x$. As before, we are ignoring the smaller rectangular prism because its volume $3dx^2$ contains $dx^2$.

> 

Reviewer: ```suggestion

In either case, the upshot is that multiplying a function by $3$ also multiplies its derivative by $3$. And of course, this goes for any constant, not just $3$. In practice, this fact is simple enough to memorize, and you'll use it thousands of times in your relationship with calculus, but every now and then it's nice to step back and reflect on why it's true and what it means in a particular context.

```

</Question>

### Derivative of $f(x) = \frac{1}{x}$

> As another example, think of the function $f(x) = \frac{1}{x}$. Now, on the one hand, you could blindly try applying the power rule, since $\frac{1}{x}$ is the same as writing $x^{-1}$. That would involve letting that $-1$ hop down, leaving behind one less than itself, $-2$. 

Reviewer: ```suggestion
As another example, think of the function $f(x) = \frac{1}{x}$. Now, on the one hand, you could blindly try applying the power rule, since $\frac{1}{x}$ is the same as writing $x^{-1}$. That would involve letting that $-1$ hop down to become a coefficient, leaving behind one less than itself in the exponent, $-2$. 
```

Reviewer: Would it be worth adding a small illustration of what's meant by "hopping down" here? Or is it pretty clear?

Author: I added an animation to the power rule figure.

But let’s have some fun and see if we can reasonable this geometrically, rather than just plugging it through a formula.

> The value $1/x$ is asking “what number multiplied by $x$ equals $1$”, so here’s how I’d visualize it: Imagine a little rectangular puddle of water in two dimensions with area $1$. Let’s say that it’s width is $x$, which means the other must be $1/x$, since the total area is $1$.

Reviewer: ```suggestion
The value $1/x$ is asking “what number multiplied by $x$ equals $1$”, so here’s how I’d visualize it: Imagine a little rectangular puddle of water in two dimensions with area $1$. Let’s say that it’s width is $x$, which means its height must be $1/x$, since the total area is $1$.
```

<Figure
  image="figures/10.58.svg" 
/>

This is a nice way to think about the graph of $1/x$, by the way. If you think of the width $x$ of this puddle in the $xy$-plane, the corresponding output $1/x$, the height of the graph above that point, is whatever height the puddle must have to maintain an area of $1$.

<Figure
  image="figures/11.00.svg"
  video="figures/11.11-11.24-trimmed.mp4" 
/>

For the derivative, imagine nudging the input $x$ up by a value $dx$. How must the height of this rectangle change so that the area remains unchanged at $1$? That is, increasing the width by $dx$ adds some new area to the right here, so the puddle must decrease in height by some $d(1/x)$ so that the area lost off the top here cancels that out.

<Figure
  image="figures/11.41.svg" 
/>

You should think of that $d(1/x)$ as being some tiny *negative* value, since it’s *decreasing* the height of this rectangle. And you know what? I’m gonna leave the last few steps here for you to pause and ponder. And once you work out $d(1/x)/dx$, compare it to what happens if you apply the power rule purely symbolically to $x^{-1}$.

$$

> \frac{d(1 / x)}{d x}= ? ? ?

Reviewer: This is actually one we should explain.  Let's frame it as a multiple-choice question.

In the explanation, mention that the new area introduced by increasing the width is dx * (1 / x), and that the change in area caused by the reduction from the top is -x * d(1 / x).  Rearrange to find that d(1 / x) = (-1 / x^2) * dx.

Also mention that the little corner of area d(1 / x) * dx in the upper right has negligible influence as $dx \to 0$

$$

> ## Comprehensive Questions

Reviewer: ```suggestion
## Exercises
```

Here are a couple questions to test your knowledge of derivatives and the past chapters.

### Derivative of $f(x) = \sqrt{x}$

> Modeling the derivatives geometrically can be applied to many different types of functions. For example, see if you can reason your way through the derivative of $f(x) = \sqrt{x}$ which also can be written as $x^{\frac{1}{2}}$. Now, one way to find the derivative of this function is to recognize that it fits the mold of the power rule. Blindly applying the rule will produce an answer, but what does it *mean*?

Reviewer: ```suggestion
See if you can reason your way through the derivative of $f(x) = \sqrt{x}$ which also can be written as $x^{\frac{1}{2}}$. By far the easiest way to compute this is to apply the power rule: The exponent hops down as a coefficient, leaving behind $\frac{1}{2} - 1 = -\frac{1}{2}$ in the exponent.

$$
\frac{df}{dx} = \frac{1}{2}x^{-1 / 2}
$$

But is this valid? And following our current playful and geometric spirit, is there a way to read what this really *means*?
```

> To start, try modeling function using geometry.

Reviewer: ```suggestion
Approaching this question with geometry is, by far, overkill. 
Frankly, it's a bit of a mind warp. 
But it does offer a satisfying explanation of an otherwise mostly symbolic fact, and more than that it's one more opportunity to flex our muscles in reasoning about how small nudges to one value can affect another.
```

<Figure
  image="figures/12.34.svg"
  caption="Unlike in previous problems, $dx$ does not represent geometric length, but instead represents **area**."
/>

<Question

>   question="What is the derivative of the function $f(x) = \sqrt{x}$?"

Reviewer: ```suggestion
  question="Using the diagram above, what is the derivative of the function $f(x) = \sqrt{x}$?"
```

  choice1="$\sqrt{2}x$"
  choice2="$\frac{1}{2\sqrt{x}}$"
  choice3="$\frac{x}{\sqrt{2}}$"
  choice4="$\frac{1}{2}x$"
  answer={2}
>

> In order to find the derivative of the function $f(x) = \sqrt{x}$ we can model its geometry as the area of the square, where given the area $x$ the function returns the length of the square's side which is the square root of $x$.

Reviewer: ```suggestion
Previously, to reason about $f(x) = x^2$, we thought of the input of the function as the side length of a square with the output being that square's area.
Here, for the function $f(x) = \sqrt{x}$, we do the opposite. We think of the input $x$ as being the area of a square, with the output $\sqrt{x}$ being the side length of that square.
```

$$
x = \sqrt{x} \sqrt{x} \\
$$

Then, we want to find $d\sqrt{x}$, the change of the function's output, when the input $x$ is nudged by a small value $dx$. Since $x$ represents area, it makes sense that a small nudge $dx$ should also be area! 

<Figure
  image="figures/12.34-explanation.svg"
  caption="Unlike in previous problems, $dx$ is not geometric length, but instead represents **area**."
/>

> The added area changes the output of the function by the new length $d\sqrt{x}$, which looks like this.

Reviewer: So the question is, how much does the side length of that square increase if the total change to the area is meant to be $dx$?

$$
x + dx = (\sqrt{x} + d\sqrt{x})(\sqrt{x} + d\sqrt{x})
$$

To find the derivative, $\frac{d\sqrt{x}}{dx}$, expand out the area expressions on the right-hand side.

$$
x + dx = x + 2\sqrt{x}(d\sqrt{x}) + (d\sqrt{x})^2
$$

Since we are only interested in the change in area, we can subtract $x$ from both sides. Additionally, we can safely ignore the expression $(d\sqrt{x})^2$ since the area it represents changes the output (length) of the function in a negligible way.

$$
dx = 2\sqrt{x}(d\sqrt{x})
$$

> Finally, we can divide both sides by $dx$ and $2\sqrt{x}$ to get the expression below. The derivative of $f(x) = \sqrt{x}$!

Reviewer: ```suggestion
Finally, we can divide both sides by $dx$ and $2\sqrt{x}$ to get the expression below.
```

$$
\frac{d\sqrt{x}}{dx}  = \frac{1}{2\sqrt{x}}
$$

It's always a good idea to check our work, and luckily we have derived such a tool to do that, the power rule. Starting with $\displaystyle f(x) = x^{\frac{1}{2}}$ we can drop the $\frac{1}{2}$ down in front and subract one from the exponent, giving us the same answer.

$$
\frac{d\sqrt{x}}{dx}  = \frac{1}{2} x^{-\frac{1}{2}}
$$

</Question>

### Volume of Sphere

In the first chapter, we found a formula for the area of a circle using a small nudge to the radius of the circle. Importantly, we derived the formula by finding the area under a function. In future chapters, we will explore this idea much further and derivatives have a big role to play. For now, let's revisit the same idea using the geometry of a sphere.

<Figure
  image="figures/sphere.svg" 
/>

> <Question

Reviewer: We should actually move this whole question much later to the chapter on integrals. It's not as relevant here, I think the sqrt question would be a fine place to end it.

Author: Sounds great, I had a feeling it was a bit out of place.
